ofrak_core/ofrak/gui/server.py

```diff
@@ -410,39 +458,77 @@ async def queue_patch(self, request: Request) -> Response:
         start = int(start_param) if start_param is not None else 0
         end_param = request.query.get("end")
         end = int(end_param) if end_param is not None else (await resource.get_data_length())
-
+        # TODO: There has to be a better way
+        new_data_string = "\\x" + "\\x".join(
+            [new_data.hex()[i : i + 2] for i in range(0, len(new_data.hex()), 2)]
+        )
+        script_str = (
+            r"""
+        {resource}.queue_patch"""
+            rf"""(Range({start}, {end}), b"{new_data_string}")"""
```

Comment:
The `r` preceding the string is not necessary. And the comment modifier does ranges using hex, so this should probably be consistent with that. In fact, any generated script ranges should be consistently using hex offsets. Please do a pass over the generated script strings looking for places with `Range` that may not be using hex, but should.

```suggestion
            f"""(Range({hex(start)}, {hex(end)}), b"{new_data_string}")"""
```